src/event_handlers/scans.rs

```diff
@@ -156,6 +157,24 @@ impl ScanHandler {
                     self.ordered_scan_url(vec![target], ScanOrder::Initial)
                         .await?;
                 }
+                Command::ModifyScanLimit(limit) => {
+
+                    let available = self.limiter.available_permits();
+                    
+                    match limit.cmp(&available){
+                        Ordering::Greater => {
+                            self.limiter.add_permits((limit - available) as usize)
+                        }
+                        Ordering::Less => {
+                            self.limiter.acquire_many((available - limit) as u32).await?.forget();
+                        }
+                        Ordering::Equal => ()
+                    }
+
+                    for scan in self.data.get_active_scans() {
+                        scan.reacquire_permit(self.limiter.clone()).await?;
+                    }
+                }
```

Comment:
add or shrink available permits to new limit. force all scans to reaquire permit, effectivley pausing if they don't fit inside of limit?